src/MoBi.Presentation/Tasks/ObservedDataTask.cs

```diff
@@ -64,26 +69,28 @@ public class ObservedDataTask : OSPSuite.Core.Domain.Services.ObservedDataTask,
 
       public void AddObservedDataToProject()
       {
-         //maybe the formats are not being registered 
          var data = _dataImporter.ImportDataSets(createMetaData().ToList(), createColumnInfos().ToList(), createDataImportSettings());
+
+         if (data.DataRepositories == null || data.Configuration == null) return;
+
          foreach (var repository in data.DataRepositories)
          {
-            adjustMolWeight(repository);
```

Comment:
Reintroduced and renamed to adjustMolWeightUnit. 

This was probably removed because of a confusion caused bz the existence of a function with the same name (adjustMolWeight()) in PK-Sim, that actually adjusted the value of the MolWeight according to the selected Molecule. Renamed the function here in MoBi to better describe what the function does and also to avoid such a confusion in the future.